### DISCUSSION TITLE
Infilling the data

### DISCUSSION THREAD

**Discussion by gabocampo** [IDEMSInternational/R-Instat#6961] [Q&A]
Hi. I have been studying the R-instat for its application in a research project, I know that R-instat can omit data, but my question is if the software can in filling in missing data of rain, if possible, How could I do it? I have watched the videos and read the guide, but it is still not clear to me.

**Comment by rdstern** [comment]
We do have a dialogue namely `Climatic > Check Data > Fill Missing Values`, but, you are right that there is hardly anything yet in the documentation or videos about its use.  This dialogue is still "work in progress" and I would be very nervous about using it currently, certainly for rainfall data.  You don't say what rainfall data - if monthly, then perhaps better, but I assume you would like to infill the daily records.  

The problem with the daily data is that it is often zero - i.e. a dry day and I assume you wish to preserve this sort of property when you infill.  That's why it is tricky.  certainly for rainfall data.  You don't say what rainfall data - if monthly, then perhaps better, but I assume you would like to infill the daily records.  

The problem with the daily data is that it is often zero - i.e. a dry day and I assume you wish to preserve this sort of property when you infill.  That's why it is tricky.  

So, what can you do now.  That depends on what other data you have.  If it is just a single station and there are just odd missing values, then you could use `Climatic > Check Data > Fill Missing Values > Fit Single > Typical` with the option `Sample`, and then the factor for the Months say.  This will sample randomly using other data values from the station in the same month.

Better if you have multiple stations, so there are other stations nearby that have a similar pattern of rainfall.  We don't have a dialogue for this, but you then choose the one that is closest in rainfall pattern and slot those values in.  (That would simply be a merge operation in the current R-Instat, if you have such a station.

We would like (in the future) to be able to use satellite-based data to do this, e.g. chirps or tamsat, but haven't yet done enough work on how the patterns of data compare.  And we expect the solution, using this idea, to be to some extent location specific.
